prow/cmd/jenkins-operator/Dockerfile

```diff
@@ -12,10 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM alpine:3.6
+FROM gcr.io/k8s-prow/alpine
```

Comment:
These don't have any sort of version tag on them. How will we bump to 3.7 when it's time?

Reply:
I'm happy to put in a 3.6, but wasn't sure what you'd want for git?

Reply:
We can version the images separately from both alpine and git. I'm mainly afraid of confusions that come about when you rely on `:latest`.

Reply:
There is a parallel here to what's called out in "Building images" [here](https://blog.nelhage.com/post/kubernetes/).

I was talking to someone about this this morning, and frankly it's a weakness `docker_build` does not share because the base image intermediate artifact is expressed declaratively, but via an explicit def/use.

In the case of that post, you'd express `FROM` as:
```python
docker_build(
   ...
  base = "//livegrep/base",
   ...
)
```

This is even easier with `lang_image` because it is containerization that "just works" (once you get Bazel to work), in their case they mention C++ and Go:
```python
cc_image(   # same signature as cc_binary
   ...
  base = "//livegrep/base",  # doesn't have to be distroless!
  ...
)

go_image(   # same signature as go_binary
   ...
  base = "//livegrep/base",  # doesn't have to be distroless!
   ...
)
```

Once we get these base images published, my next PR was going to be to give you guys `go_image` with these base images, after which I can give you `rules_k8s` for the starter cluster.

Reply:
Neat. Does it statically compile the go binary? Alpine doesn't have glibc and that can cause issues.

Reply:
No, but now that I see `go_binary` has added support for static linking I can add support for that.  By default we use `gcr.io/distroless/base` which has `glibc`, `ca-certs` and a handful of other generally necessary things (e.g. `tzdata`, a `/tmp` directory, ...).

Perhaps I will add a `:foo.static` target that does what `go_image` does, but includes the static binary version of the image ([this is how](https://github.com/bazelbuild/rules_go/pull/617) `rules_go` does this).

Reply:
Ugh, [nevermind](https://github.com/bazelbuild/rules_go/blob/master/go/modes.rst#building-static-binaries).  Nothing is easy.  (╯°□°)╯︵ ┻━┻

Reply:
Alright, I'm just going to add `build --features=static` and `run --features=static` to `.bazelrc` :)

Reply:
Alright, I sent a PR to make Bazel build static Go binaries in this repo (needs an update to `rules_go`).  That should unblock putting these binaries onto the Alpine base images.